src/libraries/System.IO.FileSystem/src/System/IO/FileStatus.Unix.cs

```diff
@@ -39,19 +39,23 @@ internal struct FileStatus
         internal bool IsReadOnly(ReadOnlySpan<char> path, bool continueOnError = false)
         {
             EnsureStatInitialized(path, continueOnError);
+            return IsReadOnly(_fileStatus);
```

Comment:
Why do we re-read if we have Refresh() for this.

Reply:
I'm not sure I understand the question.

The functionality of this code has not been changed. I merely moved the contents of the old `IsReadOnly` method into its own internal helper static method that can be accessed by both `FileSystemEntry` and `FileStatus`, to avoid repeating code.

I don't think we should do a `Refresh` call here, that's what `EnsureStatInitialized` is for - it will only perform a `Refresh` if we haven't called it yet.